### ISSUE TITLE
failed to load 'densenet_onnx' version 1: Invalid argument: unknown platform 'onnxruntime_onnx'

### ISSUE THREAD

**Issue by raghu04012012** [triton-inference-server/server#2179]
Initializing Triton Inference Server
I1027 16:50:45.049482 129238 server.cc:121]   id: 'triton'
I1027 16:50:45.049489 129238 server.cc:122]   version: '2.3.0'

-----------------------------------------------------------------------------------

Appreciate any help.

AsyncLoad() 'densenet_onnx'
I1027 16:50:45.050504 129238 model_repository_manager.cc:657] TriggerNextAction() 'densenet_onnx' version 1: 1
I1027 16:50:45.050513 129238 model_repository_manager.cc:695] Load() 'densenet_onnx' version 1
I1027 16:50:45.050520 129238 model_repository_manager.cc:714] loading: densenet_onnx:1
I1027 16:50:45.050570 129238 model_repository_manager.cc:767] CreateInferenceBackend() 'densenet_onnx' version 1
E1027 16:50:45.050672 129238 model_repository_manager.cc:899] failed to load 'densenet_onnx' version 1: Invalid argument: unknown platform 'onnxruntime_onnx'
I1027 16:50:45.050700 129238 model_repository_manager.cc:657] TriggerNextAction() 'densenet_onnx' version 1: 0
I1027 16:50:45.050708 129238 model_repository_manager.cc:672] no next action, trigger OnComplete()
I1027 16:50:45.050725 129238 model_repository_manager.cc:492] VersionStates() 'densenet_onnx'
I1027 16:50:45.050743 129238 server.cc:213] Waiting for in-flight requests to complete.
I1027 16:50:45.050751 129238 model_repository_manager.cc:595] AsyncLoad() 'densenet_onnx'
I1027 16:50:45.050762 129238 model_repository_manager.cc:657] TriggerNextAction() 'densenet_onnx' version 1: 2
I1027 16:50:45.050769 129238 model_repository_manager.cc:734] Unload() 'densenet_onnx' version 1
I1027 16:50:45.050783 129238 model_repository_manager.cc:435] LiveBackendStates()
I1027 16:50:45.050793 129238 server.cc:228] Timeout 30: Found 0 live models and 0 in-flight non-inference requests
error: creating server: Internal - failed to load all models

**Comment by GuanLuo** [comment]
Are you building the server from source or you obtain it from NGC container? If building from source, you would want to make sure it is built with `--backend=onnxruntime:<container tag>`

**Comment by spacedragon** [comment]
I found the same problem.
From code here https://github.com/triton-inference-server/server/blob/master/src/core/model_repository_manager.cc
I guess `--backend=custom` is also needed, is that correct?

**Comment by GuanLuo** [comment]
Yes, that appears to be the case, this is not expected and we will push a fix. But before that, the WAR will be specifying `--backend=custom`

**Comment by raghu04012012** [comment]
Yes building the server from source using cmake not using containers

**Comment by raghu04012012** [comment]
Backend was built by enabling TRITON_ENABLE_ONNXRUNTIME=ON

**Comment by GuanLuo** [comment]
@raghu04012012 You can try with the fix #2191

**Comment by raghu04012012** [comment]
I did try i see same issue. What other configuration options have to be specified apart from TRITON_ENABLE_ONNXRUNTIME=ON. I am building to run on CPU and i have only enabled I did try i see same issue. What other configuration options have to be specified to execute densenet example model?

**Comment by GuanLuo** [comment]
Can you provide repro steps? What cmake config you used and what command you used to run the server.

The below is the config I used (disabling most of the features except ONNX Runtime):
`cmake -DCMAKE_BUILD_TYPE=Release         -DTRITON_ENABLE_METRICS=OFF         -DTRITON_ENABLE_METRICS_GPU=OFF         -DTRITON_ENABLE_GCS=OFF         -DTRITON_ENABLE_S3=OFF         -DTRITON_ENABLE_ENSEMBLE=OFF         -DTRITON_ENABLE_CUSTOM=OFF         -DTRITON_ENABLE_TENSORRT=OFF         -DTRITON_ENABLE_TENSORFLOW=OFF -DTRITON_ENABLE_ONNXRUNTIME=ON -DTRITON_ENABLE_ONNXRUNTIME_TENSORRT=OFF         -DTRITON_ENABLE_ONNXRUNTIME_OPENVINO=OFF         -DTRITON_ENABLE_PYTORCH=OFF         -DTRITON_ENABLE_GPU=ON         -DTRITON_ENABLE_GRPC=ON         -DTRITON_ENABLE_HTTP=ON         /workspace/build`

**Comment by raghu04012012** [comment]
All the above settings are correct except i have TRITON_ENABLE_GPU=OFF

**Comment by GuanLuo** [comment]
That is also working for me.. Are you building on top of the fix commit?

**Comment by raghu04012012** [comment]
I resolved the issue but after i merge the changes suggested  the backend isnt being built

 failed to load 'densenet_onnx' version 1: Invalid argument: unable to find 'libtriton_onnxruntime.so' for model 'densenet_onnx', searched: ./model_repo/densenet_onnx/1, ./model_repo/densenet_onnx, /opt/tritonserver/backends/onnxruntime

Any suggestions as to why the backend isnt being built.

**Comment by ApoorveK** [comment]
@GuanLuo please update the current documentation, to use custom backends ( for example: onnxruntime). Documentation :  https://github.com/triton-inference-server/server/blob/main/docs/customization_guide/compose.md

**Comment by GuanLuo** [comment]
@ApoorveK can you elaborate on what to be updated in the linked documentation? This issue and fix occurred in transition to different backend management and should no longer be relevant

**Comment by ApoorveK** [comment]
@GuanLuo the current documentation actually need some examples to include cases of certain arguments like "--image", "backend" for the compose file, as those are not actually covered in the above link. That would be helpful actually
